src/relation.rs

```diff
@@ -0,0 +1,151 @@
+use pyo3::prelude::*;
+use qrlew::relation::Variant;
+use qrlew::{relation, ast};
+use serde_json::Value;
+use std::rc::Rc;
+use crate::{error, dataset};
+
+
+#[pyclass(unsendable)]
+pub struct Relation(pub Rc<relation::Relation>);
+
+#[pymethods]
+impl Relation {
+    pub fn parse(&self, query: &str, dataset: &dataset::Dataset) -> error::Result<Self> {
```

Comment:
I would call this method taking a string and returning a relation `sql` like in spark or align with the name in `Dataset`.